x/stake/keeper/delegation.go

```diff
@@ -24,41 +24,164 @@ func (k Keeper) GetDelegation(ctx sdk.Context,
 	return delegation, true
 }
 
-// load all delegations used during genesis dump
+// return all delegations used during genesis dump
 func (k Keeper) GetAllDelegations(ctx sdk.Context) (delegations []types.Delegation) {
 	store := ctx.KVStore(k.storeKey)
 	iterator := sdk.KVStorePrefixIterator(store, DelegationKey)
 
-	i := 0
-	for ; ; i++ {
-		if !iterator.Valid() {
-			break
-		}
+	for ; iterator.Valid(); iterator.Next() {
 		delegation := types.MustUnmarshalDelegation(k.cdc, iterator.Key(), iterator.Value())
 		delegations = append(delegations, delegation)
-		iterator.Next()
 	}
 	iterator.Close()
 	return delegations
 }
 
-// load all delegations for a delegator
-func (k Keeper) GetDelegations(ctx sdk.Context, delegator sdk.AccAddress,
-	maxRetrieve int16) (delegations []types.Delegation) {
+// Return all validators that a delegator is bonded to. If maxRetrieve is supplied, the respective amount will be returned.
+func (k Keeper) GetDelegatorBechValidators(ctx sdk.Context, delegatorAddr sdk.AccAddress,
```

Comment:
Remove this function - for many reasons outlined already in my PR comments

Reply:
@rigelrozanski I still need to iterate through the validators to change them to `BechValidators`. Do you prefer to do this iteration on the `Querier` or move this keeper to the `query_utils.go` file ?